src/Teambuilder/damagecalc.cpp

```diff
@@ -0,0 +1,854 @@
+    ui->ospatkstat->setText(QString::number(spatk));
+    ui->ospdefstat->setText(QString::number(spdef));
+    ui->ospdstat->setText(QString::number(spd));
+}
+
+void DamageCalc::updateMoveInfo()
+{
+    int move = MoveInfo::Number(ui->move->currentText());
+    int bp = MoveInfo::Power(move, m_currentGen);
+    int type = MoveInfo::Type(move, m_currentGen);
+    int category = MoveInfo::Category(move, m_currentGen);
+
+    ui->movepower->setValue(bp);
+    ui->movetype->setCurrentIndex(type);
+    ui->movecategory->setCurrentIndex(category);
+}
+
+void DamageCalc::updateMyPokePic()
+{
+    QPixmap p = PokemonInfo::Picture(PokemonInfo::Number(ui->mypoke->currentText()), m_currentGen, GenderInfo::Number(ui->mygender->currentText()));
+    //ui->mypokepic->setMinimumSize(qMax(64, p.width()), qMax(64, p.height()));
+    ui->mypokepic->setPixmap(p);
+}
+
+void DamageCalc::updateOPokePic()
+{
+    QPixmap p = PokemonInfo::Picture(PokemonInfo::Number(ui->opoke->currentText()), m_currentGen, GenderInfo::Number(ui->ogender->currentText()));
+    //ui->opokepic->setMinimumSize(qMax(64, p.width()), qMax(64, p.height()));
+    ui->opokepic->setPixmap(p);
+}
+
+void DamageCalc::updateMyPokeInfo()
+{
+    ui->mygender->clear();
+
+    int avail = PokemonInfo::Gender(PokemonInfo::Number(ui->mypoke->currentText()));
+
+    if (avail == Pokemon::MaleAvail) {
+        ui->mygender->addItem(tr("Male"));
+    } else if (avail == Pokemon::FemaleAvail) {
+        ui->mygender->addItem(tr("Female"));
+    } else if (avail == Pokemon::NeutralAvail) {
+        ui->mygender->addItem(tr("Neutral"));
+    } else { // male and female
+        ui->mygender->addItem(tr("Male"));
+        ui->mygender->addItem(tr("Female"));
+    }
+
+}
+
+void DamageCalc::updateOPokeInfo()
+{
+    ui->ogender->clear();
+
+    int avail = PokemonInfo::Gender(PokemonInfo::Number(ui->opoke->currentText()));
+
+    if (avail == Pokemon::MaleAvail) {
+        ui->ogender->addItem("Male");
+    } else if (avail == Pokemon::FemaleAvail) {
+        ui->ogender->addItem("Female");
+    } else if (avail == Pokemon::NeutralAvail) {
+        ui->ogender->addItem("Neutral");
+    } else { // male and female
+        ui->ogender->addItem("Male");
+        ui->ogender->addItem("Female");
+    }
+}
+
+void DamageCalc::updateMyPoke()
+{
+    updateMyPokePic();
+    updateMyPokeInfo();
+    updateMyPokeStats();
+}
+
+void DamageCalc::updateOPoke()
+{
+    updateOPokePic();
+    updateOPokeInfo();
+    updateOPokeStats();
+}
+
+void DamageCalc::calculate()
+{
+    // INFO // http://www.smogon.com/bw/articles/bw_complete_damage_formula
+
+    int move = MoveInfo::Number(ui->move->currentText());
+    int bp = ui->movepower->value();
+    int type = TypeInfo::Number(ui->movetype->currentText());
+    int category = ui->movecategory->currentIndex();
+    bool crit = ui->crit->isChecked();
+    int gen = m_currentGen.num;
+    int weather = ui->weather->currentData().toInt();
```

Comment:
Weather differs between gens and its not in order of gens introduced, so that doesnt work